RFC-0006-nan-definition.md

```diff
@@ -0,0 +1,213 @@
+https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL 
+
+#### Spark SQL
+
+There is special handling for not-a-number (NaN) when dealing with float or double types that does not exactly match standard floating point semantics. Specifically:
+NaN = NaN returns true.
+In aggregations, all NaN values are grouped together.
+NaN is treated as a normal value in join keys.
+NaN values go last when in ascending order, larger than any other numeric value. (this holds for min/max functions as well, such as array_max)
+ https://spark.apache.org/docs/3.0.0-preview/sql-ref-nan-semantics.html
+
+#### Snowflake
+
+In Snowflake, NaN=NaN and is sorted as greater than infinity: https://docs.snowflake.com/en/sql-reference/data-types-numeric#special-values 
+
+#### BigQuery
+
+NaN is considered smaller than -infinity for sorting. 
+All NaN values are considered equal for grouping and distinct operations
+min/max functions return NaN if NaN is in the input
+BigQuery does not have functions that do distincts or sorting on arrays or maps
+https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types 
+https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions#max
+
+#### MySQL
+
+Mysql doesn't support NaNs at all (see reference here https://dev.mysql.com/doc/refman/8.0/en/type-conversion.html)
+
+#### Oracle
+Oracle orders NaN greatest with respect to all other values, and evaluates NaN as equal to NaN.
+
+https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Data-Types.html#GUID-33A52FDB-BA5C-474E-96D3-40390BA5F5F4
+
+#### Db2 for z/os and Db2 LUW
+
+DB2 has two kinds of floating point numbers.  The first, DOUBLE or FLOAT, does not support NaN or infinity values.  The second, DECFLOAT, supports infinities as well as quiet and signaling NaNs. It is described as an IEEE-754 number, but does not follow the IEEE spec for comparisons.  The NaNs compare as equal to each other and greater in magnitude than any other number. The ordering among the different special values is as follows: -NAN < -SNAN < -INFINITY < 0 < INFINITY < SNAN <NAN.
+
+https://www.ibm.com/docs/en/db2-for-zos/13?topic=numbers-double-precision-floating-point-double-float
+https://www.ibm.com/docs/en/db2-for-zos/13?topic=numbers-decimal-floating-point-decfloat
+https://www.ibm.com/docs/en/db2-for-zos/13?topic=comparison-numeric-comparisons 
+
+https://www.ibm.com/docs/en/db2/11.5?topic=list-numbers#r0008469__title__7
+https://www.ibm.com/docs/en/db2/11.5?topic=list-numbers#r0008469__title__8
+https://www.ibm.com/docs/en/db2/11.5?topic=types-assignments-comparisons#r0008479__numcomp__title__1
+## Goals
+
+* Define a consistent behavior for ordering and comparisons of NaNs in Presto
+* Make behavior easy to implement and debug
+* Make behavior easy for users to understand
+* Align behavior with Velox for consistency across Java and C++ workers
```

Comment:
Add Align with Spark?

Reply:
I think that's a nice to have when choosing between options, but not a primary goal.